src/libraries/System.Diagnostics.DiagnosticSource/tests/MetricEventSourceTests.cs

```diff
@@ -1290,36 +1292,40 @@ public void EventSourcePublishesEndEventsOnMeterDispose()
-            Histogram<int> h = meterB.CreateHistogram<int>("histogram1", "a unit", "the description", new TagList() { { "hk1", "hv1" }, { "hk2", "" }, {"hk3", null } });
-            UpDownCounter<int> udc = meterA.CreateUpDownCounter<int>("upDownCounter1", "udc unit", "udc description", new TagList() { { "udk1", "udv1" } });
-            int upDownCounterState = 0;
-            ObservableUpDownCounter<int> oudc = meterA.CreateObservableUpDownCounter<int>("observableUpDownCounter1", () => { upDownCounterState += 11; return upDownCounterState; }, "oudc unit", "oudc description");
+                object scope = new object();
 
-            EventWrittenEventArgs[] events;
-            using (MetricsEventListener listener = new MetricsEventListener(_output, MetricsEventListener.InstrumentPublishing, null, ""))
-            {
-                listener.WaitForEnumerationComplete(s_waitForEventTimeout);
-                events = listener.Events.ToArray();
-            }
+                using Meter meterA = new Meter("TestMeter10", null, null, scope);
+                using Meter meterB = new Meter("TestMeter11", null, new TagList() { { "Mk1", "Mv1" }, { "Mk2", null } }, scope);
+                Counter<int> c = meterA.CreateCounter<int>("counter1", "hat", "Fooz!!");
+                Gauge<int> g = meterA.CreateGauge<int>("gauge1", "C", "Temperature");
+                int counterState = 3;
+                ObservableCounter<int> oc = meterA.CreateObservableCounter<int>("observableCounter1", () => { counterState += 7; return counterState; }, "MB", "Size of universe",
+                                                new TagList() { { "ock1", "ocv1" }, { "ock2", "ocv2" }, { "ock3", "ocv3" } });
+                int gaugeState = 0;
+                ObservableGauge<int> og = meterA.CreateObservableGauge<int>("observableGauge1", () => { gaugeState += 9; return gaugeState; }, "12394923 asd [],;/", "junk!",
+                                                new TagList() { { "ogk1", "ogv1" } });
+                Histogram<int> h = meterB.CreateHistogram<int>("histogram1", "a unit", "the description", new TagList() { { "hk1", "hv1" }, { "hk2", "" }, {"hk3", null } });
+                UpDownCounter<int> udc = meterA.CreateUpDownCounter<int>("upDownCounter1", "udc unit", "udc description", new TagList() { { "udk1", "udv1" } });
+                int upDownCounterState = 0;
+                ObservableUpDownCounter<int> oudc = meterA.CreateObservableUpDownCounter<int>("observableUpDownCounter1", () => { upDownCounterState += 11; return upDownCounterState; }, "oudc unit", "oudc description");
 
-            AssertInstrumentPublishingEventsPresent(events, c, oc, og, h, udc, oudc, g);
-            AssertInitialEnumerationCompleteEventPresent(events);
+                EventWrittenEventArgs[] events;
+                using (MetricsEventListener listener = new MetricsEventListener(NullTestOutputHelper.Instance, MetricsEventListener.InstrumentPublishing, null, "*"))
```

Comment:
Is the change from "" -> "*" required to make the test work? If so that suggests we've created an unintended breaking change in the MetricsEventSource syntax and ideally we'd undo that break.

cc @CodeBlanch

Reply:
I can fix this allowing empty string behave as "*" listening to all instruments. Yes, if we don't do that there will be breaking change. It took me some time trying to figure out what was the problem.

Reply:
I have added the fix to allow empty string to listen to all instruments as we used to do before.